gwpy/signal/spectral/_scipy.py

```diff
@@ -200,6 +206,35 @@ def coherence(timeseries, other, segmentlength, noverlap=None, **kwargs):
     --------
     scipy.signal.coherence
     """
+
+    # Should we warn about unequal sampling frequencies?
+    warn_fs = False
+
+    if downsample is None:
+        warn_fs = True
+        downsample = True
+
+    if timeseries.sample_rate != other.sample_rate:
+        # scipy assumes a single sampling frequency
+        if not downsample:
+            raise ValueError("Cannot calculate coherence when sampling "
+                             "frequencies are unequal")
+        else:
+            if warn_fs:
+                warnings.warn("Sampling frequencies are unequal. Higher "
+                              "frequency series will be downsampled before "
+                              "coherence is calculated",
+                              category=UserWarning)
+                # downsample the one with the higher rate
+                if timeseries.sample_rate > other.sample_rate:
+                    timeseries = timeseries.resample(other.sample_rate)
+                else:
+                    other = other.resample(timeseries.sample_rate)
```

Comment:
This block is currently inside the `warn_fs` block, is tht right? Or should this be

```suggestion
            # downsample the one with the higher rate
            if timeseries.sample_rate > other.sample_rate:
                timeseries = timeseries.resample(other.sample_rate)
            else:
                other = other.resample(timeseries.sample_rate)
```

Reply:
Thanks, yes, fixed and added a test to account for the resulting problem